### ISSUE TITLE
Calls to template paths don't play well with child themes

### ISSUE THREAD

**Issue by helion** [tammyhart/Reusable-Custom-WordPress-Meta-Boxes#1]
When a child theme is being used the constant TEMPLATEPATH and the function calls get_bloginfo('template_directory') and get_template_directory_uri(), all return the path to a parent theme and not the active child theme.

Part of the value of child themes is to use and customize theme frameworks while still leaving them easy to upgrade. Relying on function that call the template path forces users of this tool to add it and the function calls that use it to the parent theme even if they are using a child theme. This makes upgrading that theme more difficult and time consuming and can force a developer to separate their custom code into two separate theme directories. 

By using different functions and constants its possible to get the path of a child theme if one is being used while getting the same value as the functions listed above if no child theme is in use.

Unless I'm missing a reason that makes the template functions more desirable than the stylesheet variants my suggestion is to do the following : 
- In functions.php replace the TEMPLATEPATH constant with either STYLESHEETPATH or get_stylesheet_directory() which is what's used to populate the constant in the wp-includes/default-constants.php file (line 295). 
- In meta_box.php on lines 9, 11, and 12 replace get_bloginfo('template_directory') with get_stylesheet_directory_uri(). The WP Codex page for get_bloginfo() seems to recommend that function instead of get_bloginfo('stlyesheed_directory')  but doesn't explain why. 
- In meta_box.php on line 160 replace get_template_directory_uri() with get_stylesheet_directory_uri()

**Comment by MarioGiancini** [comment]
helion,

Noticed this too. Thanks for finding all the lines to change.

**Comment by MarioGiancini** [comment]
Actually, looks like with the new updates I'm missing something.  The code lines you listed above have changed since last update.  Found most of them, but jquery-ui-core and jquery-ui-datepicker aren't loading with the rest of the scripts in the admin and causing some issues (checked the page source and compared to another site of mine that installed in a core theme, not child theme). Date picker isn't working, but the date field and other fields and meta box shows up fine. Also, the other boxes like custom fields, sharing, etc won't close or open when clicking the triangle in the right hand corner.  I'm assuming thats because datepick and jquery-ui-core haven't loaded.

I have the boxes loading only on a particular page template. Perhaps this might be an issue? I don't see where though...

**Comment by MarioGiancini** [comment]
So I answered my own question:

1.) If you put the metabox directory into another directory that breaks many references to load the custom js scripts and the jquery dateppicker UI stylesheet.  So it's best to put the metabox directory right in the main directory of your theme (or this case childtheme).

2.) The newest version of meta_box.php is missing the actual calls to enqueue the needed scripts for datepicker, slider and farbtastic in the date, slider, and color fields respectively.  It looks like it just adds their name to an array $deps which is used for some of the custom js and css natively in Wordpress.  For each one of those if statements I added the wp_enqueue_script() with the script name.  Wordpress already knows about the script so I didn't add name or dependents to it.  The if statements start at line 103. 

Hope that helps anyone else who might have difficulty.
